modules/platforms/cpp/core/src/ignition.cpp

```diff
@@ -149,6 +149,24 @@ namespace ignite
                 opts.push_back(CopyChars("--add-exports=java.base/sun.reflect.generics.reflectiveObjects=ALL-UNNAMED"));
                 opts.push_back(CopyChars("--add-opens=jdk.management/com.sun.management.internal=ALL-UNNAMED"));
                 opts.push_back(CopyChars("--illegal-access=permit"));
+
+                // Those are only needed for Java 15+, but Java 15 can not be detected easily using JNI,
+                // so just putting it here in case we are running on 15+. It is OK to have them on Java 9-14 too.
```

Comment:
We can add a link to https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#getversion, which shows why it is not possible to detect Java version with JNI.

Reply:
Done